dspace-api/src/main/java/org/dspace/content/enhancer/script/ItemEnhancerEntityTypeScript.java

```diff
@@ -0,0 +1,304 @@
+                    if (Objects.nonNull(this.entitytype)) {
+                        itemService.findAll(context, limit, counter).forEachRemaining(this::enhanceItemEntityCheck);
+                    } else {
+                        itemService.findAll(context, limit, counter).forEachRemaining(this::enhanceItem);
+                    }
+                    counter += limit;
+                    context.commit();
+                    handler.logInfo("processed " + counter + " out of total " + total + " items");
+                    log.info("processed " + counter + " out of total " + total + " items");
+                } catch (SQLException e){
+                    handler.logError(e.getMessage());
+                    counter += limit;
+                    handler.logInfo("processed " + counter + " out of total " + total + " items");
+                    log.info("processed " + counter + " out of total " + total + " items");
+                }
+            }
+        }
+
+    }
+
+    private Iterator<Item> findItemsToEnhance() {
+        try {
+        	Iterator<Item> result = null;
+         	if (Objects.nonNull(collection)) {
+         		//Check, if uuid exist
+        			Collection coll = collectionService.find(context, collection);
+        			if(coll != null) {
+        				result = itemService.findAllByCollection(context, coll);
+        			}
+        	} else {
+        		result = itemService.findAll(context);
+        	}
+        	return result;
+        } catch (SQLException e) {
+            throw new SQLRuntimeException(e);
+        }
+    }
+
+    private void enhanceItem(Item item) {
+        if (force) {
+            itemEnhancerService.forceEnhancement(context, item);
+        } else {
+            itemEnhancerService.enhance(context, item);
+        }
+
+        uncacheItem(item);
+
+    }
+
+    /**
+     * Additional Entity Check.
+    */
+    private void enhanceItemEntityCheck(Item item) {
+        if (Objects.nonNull(entitytype)) {
+            if (!entitytype.contentEquals(itemService.getEntityType(item))) {
+                uncacheItem(item);
+            }
+        } else {
+            if (force) {
+                itemEnhancerService.forceEnhancement(context, item);
+            } else {
+                itemEnhancerService.enhance(context, item);
+            }
+
+            uncacheItem(item);
+        }
+    }
+
+    private void uncacheItem(Item item) {
+        try {
+            context.uncacheEntity(item);
+        } catch (SQLException e) {
+            throw new SQLRuntimeException(e);
+        }
+    }
+
+    private void assignCurrentUserInContext() throws SQLException {
+        UUID uuid = getEpersonIdentifier();
+        if (uuid != null) {
+            EPerson ePerson = EPersonServiceFactory.getInstance().getEPersonService().find(context, uuid);
+            context.setCurrentUser(ePerson);
+        }
+    }
+
+    private void assignSpecialGroupsInContext() throws SQLException {
```

Comment:
Exception 'java.sql.SQLException' is never thrown in the method.

Reply:
The Exception was adopted from the existing RelatedEntityItemEnhancer and now has been removed.